project/rroykuo/README.md

```diff
@@ -0,0 +1,57 @@
+# Clip your arbitrary polygons
+
+## Basic Information
+Clipping polygons means using a polygon to clip a subject one. Only intersection parts of both would be retain. This function would be used to image processing or clipping a image into certain shape.
+
+> ![](https://i.imgur.com/jugvja9.png)
+>
+> **Clipping of polygons**
+
+## Problem to solve
+Some algorithms to solve clipping polygons might be fast but have some limit of input (ex. polygons have to be convex). Greiner–Hormann clipping algorithm can be fast and easy than other algorithm. Furthermore, it doesn't limit the shape of polygons. Therefore, I decide to reproduce the polygon clipping function with this algorithm (Greiner–Hormann clipping algorithm) in C++. Both clip and subject polygons could be concave or convex.
+
+
+## Engineer Infrastructure
+* Environment: `Ubuntu22.04`
+* Automatic build system: `cmake`
+* Version control: `git`
+* Testing framework: `pytest`
+* Documentation: `GitHub README.md`
+
+## System Archtecture
+
+> ![](https://i.imgur.com/2jn4i96.png)
+>
+> **block diagram of system**
+
+- **User input process** : I would choose `OpenGL` to display polygons and ask users to input clip and subject polygons. The interface is the window where `OpenGL` shows the result. Users decides the count of vertices of two polygons then point vertices one by one on window via mouse. So I should save the position(x, y) of vertices in each polygon and display the input polygons immediately.
```

Comment:
What is the dimensionality of the space that your polygons are defined in?

Reply:
2D dimension